### PR TITLE
Terragrunt Provider Cache with `providers lock` command

### LINKED ISSUE #3163: Terragrunt Provider Cache with multiple platforms
## Describe the bug

The current implementation of Terragrunt Provider Cache creates a cache only for the platform on which it runs. We need to improve it to be able create cache with lock file for multiple platforms. Something like 
`terraform providers lock -platform=linux_amd64 -platform=darwin_arm64`

## Versions

- Terragrunt version: v0.58.10:

## Additional context

https://github.com/gruntwork-io/terragrunt/issues/2920#issuecomment-2109327569

### PR DESCRIPTION
**Pull request by levkohimins**
## Description

Fixes #3163.

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [x] Update the docs.
- [x] Run the relevant tests successfully, including pre-commit checks.
- [ ] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [ ] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes (draft)

Added / Removed / Updated [X].

### CODE REVIEW

File: terraform/getproviders/lock.go [hunk 1]
```diff
@@ -117,6 +119,33 @@ func updateProviderBlock(ctx context.Context, providerBlock *hclwrite.Block, pro
 	return nil
 }
 
+func getExistingHashes(providerBlock *hclwrite.Block, provider Provider) ([]Hash, error) {
+	var hashes []Hash
+
+	versionAttr := providerBlock.Body().GetAttribute("version")
+	if versionAttr != nil {
```
Comment (denis256) [inline terraform/getproviders/lock.go:126]: To reduce nested `{` blocks, make sense to return `return hashes, nil` if versionAttr == nil?

File: test/integration_test.go [hunk 2]
```diff
@@ -206,6 +209,64 @@ const (
...
+
+	for _, appName := range []string{"app1", "app2", "app3"} {
+		appPath := filepath.Join(rootPath, appName)
+		assert.True(t, util.FileExists(appPath))
+
+		lockfilePath := filepath.Join(appPath, ".terraform.lock.hcl")
+		lockfileContent, err := os.ReadFile(lockfilePath)
+		require.NoError(t, err)
+
+		lockfile, diags := hclwrite.ParseConfig(lockfileContent, lockfilePath, hcl.Pos{Line: 1, Column: 1})
```
Comment (denis256) [inline test/integration_test.go:250]: makes sense to check `lockfile` for nil since it is pointer

File: terraform/getproviders/lock.go (hunk 1 above)
Comment (levkohimins) [reply terraform/getproviders/lock.go]: Indeed. Thanks for the tip.

File: test/integration_test.go (hunk 2 above)
Comment (levkohimins) [reply test/integration_test.go:250]: Agree, thanks.

**Review by denis256 — APPROVED** [review]

**Comment by levkohimins** [comment]
Resolved in [v0.58.14](https://github.com/gruntwork-io/terragrunt/releases/tag/v0.58.14) release.

**Pull request merged**